test/e2e/custom_wait_rules_test.go

```diff
@@ -223,39 +225,36 @@ metadata:
 	}()
 
 	logger.Section("deploy resource with current state as progressing", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", crdApp}, RunOpts{StdinReader: strings.NewReader(crd)})
+
 		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{
 			StdinReader: strings.NewReader(yaml)})
 
 		out = strings.TrimSpace(replaceSpaces(replaceTarget(replaceTs(out))))
 
 		expectedOutput := strings.TrimSpace(replaceSpaces(`Changes
 
-Namespace  Name                         Kind                      Age  Op      Op st.  Wait to    Rs  Ri  $
-(cluster)  crontabs.stable.example.com  CustomResourceDefinition  -    create  -       reconcile  -   -  $
-kapp-test  my-new-cron-object-1         CronTab                   -    create  -       reconcile  -   -  $
-^          test-cm                      ConfigMap                 -    create  -       reconcile  -   -  $
+Namespace  Name                  Kind       Age  Op      Op st.  Wait to    Rs  Ri  $
+kapp-test  my-new-cron-object-1  CronTab    -    create  -       reconcile  -   -  $
+^          test-cm               ConfigMap  -    create  -       reconcile  -   -  $
 
```

Comment:
why did we need to split into 2 kapp deploys? will users using kapp have to split things as well when encountering similar cases?

Reply:
Sometimes while deploying the crontab CRD, we get "Condition Established is not set" before the CRD is successfully created. This messes up the expected out. Since the main concern for this test case is the CR and the ConfigMap (CR has the wait rules defined and the ConfigMap is unblocked when it's in the Progressing state), I moved the CRD to a separate app to have a deterministic expected output.

Reply:
i see.